pkg/parse/parse.go

```diff
@@ -387,12 +387,24 @@ func GetBindMount(args []string) (specs.Mount, error) {
 		}
 	}
 
+	// buildkit parity: default bind option must be `rbind`
+	// unless specified
+	if !bindNonRecursive {
+		newMount.Options = append(newMount.Options, "rbind")
+	}
+
 	if !setDest {
 		return newMount, noDestError
 	}
 
-	if !setSource {
-		newMount.Source = newMount.Destination
+	// builkit parity: support absolute path for sources from current build context
+	if newMount.Source == "" || newMount.Source == "." || newMount.Source == "./" {
+		newMount.Source = contextDir
+	}
+	// if path begins with `./` replace `./` with context dir
+	if strings.HasPrefix(newMount.Source, "./") {
+		// path should be /contextDir/specified path
+		newMount.Source = strings.Replace(newMount.Source, ".", contextDir, 1)
```

Comment:
Sounds like `filepath.Join(contextDir, filepath.Clean(filepath.Separator+newMount.Source))` is going to do the right thing here.  We need to be careful that we don't accidentally end up binding in a location that's actually outside of the context directory.